ios/LayoutReanimation/REAUIManager.mm

```diff
@@ -85,24 +114,23 @@ - (void)_manageChildren:(NSNumber *)containerTag
 
   // Reanimated changes /start
   BOOL isUIViewRegistry = ((id)registry == (id)[self valueForKey:@"_viewRegistry"]);
-  id<RCTComponent> container;
-  NSMutableArray<id<RCTComponent>> *permanentlyRemovedChildren;
   if (isUIViewRegistry) {
-    container = registry[containerTag];
-    for (id<RCTComponent> toRemoveChild in _toBeRemovedRegister[containerTag]) {
-      [container removeReactSubview:toRemoveChild];
-    }
-
-    permanentlyRemovedChildren = (NSMutableArray *)[super _childrenToRemoveFromContainer:container
-                                                                               atIndices:removeAtIndices];
-    if (permanentlyRemovedChildren != nil) {
-      for (id<RCTComponent> permanentlyRemovedChild in permanentlyRemovedChildren) {
-        if (_toBeRemovedRegister[containerTag] == nil) {
-          _toBeRemovedRegister[containerTag] = [[NSMutableSet<id<RCTComponent>> alloc] init];
+    BOOL wasProxyRemovalSet = _proxyViewRemoval;
+    if (!_proxyViewRemoval) {
+      id<RCTComponent> container = registry[containerTag];
+      NSArray<id<RCTComponent>> *permanentlyRemovedChildren = [self _childrenToRemoveFromContainer:container
```

Comment:
Do you have some kind of formal proof that all animated views with layout animations are indeed eventually removed?

Reply:
@kmagiera ?

Reply:
not really a formal proof but after the change there is only a single code path that adds and another one that removes the views from the "exiting views" map. Another imporant factor is that we now only store references to the "exiting views" as opposed to all the views that there are. These changes makes it much more easy for this code to reason about. We can see that exiting views are only added for the tree elements that are removed with an exit animation or their parent that are removed when the animation is running. The code path that cleans that up runs on the animation callback. So assuming the end callback always run after the exit animation has started at some point, the views will be cleared up from that map.